Source/WebCore/platform/graphics/ImageFrame.h

```diff
@@ -88,13 +87,17 @@ class ImageFrame {
     bool hasDecodedNativeImageCompatibleWithOptions(const std::optional<SubsamplingLevel>&, const DecodingOptions&) const;
     bool hasMetadata() const { return !size().isEmpty(); }
 
+    bool hasCachedDecodedImage(const std::optional<SubsamplingLevel>& = { }) const;
+    bool hasCachedDecodedImageCompatibleWithOptions(const std::optional<SubsamplingLevel>&, const DecodingOptions&) const;
+
     Color singlePixelSolidColor() const;
 
 private:
     DecodingStatus m_decodingStatus { DecodingStatus::Invalid };
     IntSize m_size;
 
     RefPtr<NativeImage> m_nativeImage;
+    std::unique_ptr<CachedDecodedImage> m_cachedDecodedImage;
```

Comment:
Can an ImageFrame has valid NativeImage and CachedDecodedImage at the same time? If not should not these two members be a variant?